### ISSUE TITLE
Exception when receiving invite to non-tracked channel

### ISSUE THREAD

**Issue by theChrisliebaer** [KittehOrg/KittehIRCClientLib#259]
# Issue
Library does not handle beeing invited properly.

# How to reproduce
0. Use version `5.1.0`
1. Join to IRC network with client of your liking
2. Connect with KittehIRC library to same network
3. Use invite command to invite KittehIRC client to any channel it hasn't joined so far.
4. Library will be unable to handle invite message from server

# Log output
```
org.kitteh.irc.client.library.exception.KittehServerMessageException: Trouble processing message: INVITE message sent for invalid channel name. Messages: 
:Chrisliebaer!~Chrislieb@unaffiliated/chrisliebaer INVITE Chrisliebot2 :#chrisliebot
	at org.kitteh.irc.client.library.defaults.listener.AbstractDefaultListenerBase.trackException(AbstractDefaultListenerBase.java:142)
	at org.kitteh.irc.client.library.defaults.listener.DefaultInviteListener.invite(DefaultInviteListener.java:63)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at net.engio.mbassy.dispatch.ReflectiveHandlerInvocation.invoke(ReflectiveHandlerInvocation.java:29)
	at net.engio.mbassy.dispatch.MessageDispatcher.dispatch(MessageDispatcher.java:30)
	at net.engio.mbassy.dispatch.FilteredMessageDispatcher.dispatch(FilteredMessageDispatcher.java:42)
	at org.kitteh.irc.client.library.feature.filter.FilteringSubscriptionFactory$FilteredMessageDispatcher.dispatch(FilteringSubscriptionFactory.java:89)
	at net.engio.mbassy.subscription.Subscription.publish(Subscription.java:72)
	at net.engio.mbassy.bus.MessagePublication.execute(MessagePublication.java:49)
	at net.engio.mbassy.bus.MBassador.publish(MBassador.java:66)
	at org.kitteh.irc.client.library.defaults.feature.DefaultEventManager.callEvent(DefaultEventManager.java:130)
	at org.kitteh.irc.client.library.defaults.DefaultClient.sendLineEvent(DefaultClient.java:986)
	at org.kitteh.irc.client.library.defaults.DefaultClient.handleLine(DefaultClient.java:942)
	at org.kitteh.irc.client.library.defaults.DefaultClient.access$000(DefaultClient.java:105)
	at org.kitteh.irc.client.library.defaults.DefaultClient$InputProcessor.processElement(DefaultClient.java:167)
	at org.kitteh.irc.client.library.defaults.DefaultClient$InputProcessor.processElement(DefaultClient.java:159)
	at org.kitteh.irc.client.library.util.QueueProcessingThread.run(QueueProcessingThread.java:63)
org.kitteh.irc.client.library.exception.KittehServerMessageException: Trouble processing message: INVITE message sent for invalid channel name. Messages: 
:Chrisliebaer!~Chrislieb@unaffiliated/chrisliebaer INVITE Chrisliebot2 :#kitinfo
	at org.kitteh.irc.client.library.defaults.listener.AbstractDefaultListenerBase.trackException(AbstractDefaultListenerBase.java:142)
	at org.kitteh.irc.client.library.defaults.listener.DefaultInviteListener.invite(DefaultInviteListener.java:63)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at net.engio.mbassy.dispatch.ReflectiveHandlerInvocation.invoke(ReflectiveHandlerInvocation.java:29)
	at net.engio.mbassy.dispatch.MessageDispatcher.dispatch(MessageDispatcher.java:30)
	at net.engio.mbassy.dispatch.FilteredMessageDispatcher.dispatch(FilteredMessageDispatcher.java:42)
	at org.kitteh.irc.client.library.feature.filter.FilteringSubscriptionFactory$FilteredMessageDispatcher.dispatch(FilteringSubscriptionFactory.java:89)
	at net.engio.mbassy.subscription.Subscription.publish(Subscription.java:72)
	at net.engio.mbassy.bus.MessagePublication.execute(MessagePublication.java:49)
	at net.engio.mbassy.bus.MBassador.publish(MBassador.java:66)
	at org.kitteh.irc.client.library.defaults.feature.DefaultEventManager.callEvent(DefaultEventManager.java:130)
	at org.kitteh.irc.client.library.defaults.DefaultClient.sendLineEvent(DefaultClient.java:986)
	at org.kitteh.irc.client.library.defaults.DefaultClient.handleLine(DefaultClient.java:942)
	at org.kitteh.irc.client.library.defaults.DefaultClient.access$000(DefaultClient.java:105)
	at org.kitteh.irc.client.library.defaults.DefaultClient$InputProcessor.processElement(DefaultClient.java:167)
	at org.kitteh.irc.client.library.defaults.DefaultClient$InputProcessor.processElement(DefaultClient.java:159)
	at org.kitteh.irc.client.library.util.QueueProcessingThread.run(QueueProcessingThread.java:63)

```

Additionally, it doesn't seem like there even is any support for (incoming) invites in KittehIRC. Note that I'm talking about inviting a bot, I'm not talking about another client inviting a third party to a channel. Would be great if you guys could implement an event for that.

**Comment by mbax** [comment]
Thaaaaaaat's embarrassing, I'll try to fix that tonight.

**Comment by mbax** [comment]
Update: Yeah that is just really dumb handling of the invite. I'll have 5.1.1 out tonight, which will be 5.1.0 plus the fix. Also 6.0.0 (probably also out tonight, full of breaking changes) will also contain the fix.

**Comment by mbax** [comment]
Fixed in this commit, 5.1.1 publishing next. Took a bit longer because I decided I may as well be thorough and fix the other places it could cause trouble too.
https://github.com/KittehOrg/KittehIRCClientLib/commit/94cd8e4051c370ceda8d5ec00057b9d6dfb14a66
